src/types/exceptions.rs

```diff
@@ -342,7 +342,7 @@ impl UnicodeDecodeError {
         reason: &CStr,
     ) -> PyResult<&'p PyObjectRef> {
         unsafe {
-            let input: &[c_char] = &*(input as *const [u8] as *const [i8]);
+            let input: &[c_char] = std::mem::transmute::<&[u8], &[c_char]>(input);
```

Comment:
Please do not use transmute if not strictly required. See https://doc.rust-lang.org/nightly/std/mem/fn.transmute.html:

> transmute is incredibly unsafe. There are a vast number of ways to cause undefined behavior with this function. transmute should be the absolute last resort.

Reply:
I'm open to a different way of doing this. The fundamental issue is that `char` is unsigned by default on some platforms (like ARM) and signed by default on others (such as x86). So, this code compiles fine on x86 but results in a compilation error when compiling for ARM. I thought about using a `cfg(target_arch = xxx)` for this but that has the downside of constantly playing catch up as new platforms are added.

Reply:
Huh I didn't know about that! Would `let input: &[c_char] = &*(input as *const [u8] as *const [c_char]);` work? I've checked it works x86

Reply:
I'm incorporating your other suggestions now and will try this.